alica_dummy_proxy/src/communication/AlicaDummyCommunication.cpp

```diff
@@ -57,7 +57,10 @@ class InProcQueue
 
     ~InProcQueue()
     {
-        _isRunning = false;
+        {
+            std::unique_lock<std::mutex> lck(_qmutex);
+            _isRunning = false;
+        }
         _qcondition.notify_one();
```

Comment:
Maybe `notify_all()`?

Reply:
I really like it too, but here we have only one thread: `InProcQueue::run()`

Reply:
yeah, just seems more appropriate